lib/rules/no-array-destructured-return.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @fileoverview Disallow array destructuring of return values.
+ * @author Vitor Balocco
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "disallow array destructuring of return values",
+            category: "Best Practices",
+            recommended: false
+        },
+
+        schema: []
+    },
+
+    create(context) {
+
+        /**
+         * Reports an error on the given node.
+         * @param   {ASTNode} node The node to report on.
+         * @returns {void}
+         */
+        function report(node) {
+            context.report({
+                node,
+                message: "Array destructuring of return values is not allowed, use object destructuring instead."
```

Comment:
@platinumazure no, I don't think you are overthinking it, that's actually a pretty good point! I'm having a hard time thinking of a better alternative, though. How about something like:
`"Array destructuring of return values is not allowed. You should refactor the invoked function to return an object instead."`